src/shogun/machine/GeneralizedLinearMachine.h

```diff
@@ -0,0 +1,215 @@
+/*
+ * STILL A WORK IN PROGRESS
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Tej Sukhatme
+ */
+
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/machine/Machine.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/lib/SGMatrix.h>
+
+
+namespace shogun
+{
+    enum GLM_DISTRIBUTION
+	{
+        //TODO GAUSSIAN,
+        //TODO BINOMIAL,
+        //TODO GAMMA,
+        //TODO SOFTPLUS,
+        //TODO PROBIT,
+        POISSON
+	};
+
+class DotFeatures;
+class Features;
+class RegressionLabels;
+
+/** @brief Class for estimating regularized generalized linear models (GLM).
```

Comment:
where is this coming from? I don't see anything credited. You can't just copy paste code from somewhere else (not even docs). Always state the source, and also under what license the source had it

Reply:
Yes sure, even the docs are based on the [pyGLMnet](https://github.com/glm-tools/pyglmnet/blob/master/pyglmnet/pyglmnet.py) code (again, MIT license). However it has not been directly copy pasted.
How do I give credit? As in what is the format and where do I mention it?

Reply:
whenever you copy paste stuff, you need to reference that. And even if you first copy paste and then adapt it you should do that. Note that those docs are not necessarily appropriate here. Do we really want group lasso /elastic net style penalties? I think we just want a poisson regression

Reply:
> whenever you copy paste stuff, you need to reference that. And even if you first copy paste and then adapt it you should do that. Note that those docs are not necessarily appropriate here. Do we really want group lasso /elastic net style penalties? I think we just want a poisson regression

It would be better to have at least the LASSO regularization term imo. It is also kind of needed for the influenza project.

Reply:
you could say in the license part of the file that both code and docs are heavily inspired by pyGLMnet

Reply:
thanks @geektoni I didn't know that. 
L1 is not group lasso though ...